src/vectorstore/surrealdb_store.py

```
@@ -89,25 +96,36 @@ async def add_documents(self, documents: List[Document]) -> List[str]:
 
             doc_ids = []
 
-            # Batch insert documents
+            # PARALLEL BATCH INSERT - Create all insert tasks first
+            insert_tasks = []
             for doc, embedding in zip(documents, embeddings):
```

Ruff (0.14.14)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)